### PR TITLE
Migrate Travis to use docker to test on Xenial and Bionic

### PR DESCRIPTION
**Pull request by traversaro**

### CODE REVIEW

**Comment by traversaro** [comment]
Fix https://github.com/robotology/idyntree/issues/458

File: .ci/script.sh [hunk 1]
```diff
@@ -0,0 +1,40 @@
+#!/bin/sh
+set -e
+
```
[the remark this answers is not in the document]
Comment (traversaro) [reply .ci/script.sh]: Cool. What's the syntax ? 
~~~
set -e -u
~~~ 
or 
~~~
set -e
set -u
~~~ 
?

[the remark this answers is not in the document]
Comment (diegoferigo) [reply .ci/script.sh]: It is the same, probably even `set -eu` works

**Comment by traversaro** [comment]
I changed my mind. Let's keep the bionic build, let's just disable Valgrind on those.

**Pull request merged**